tests/unit/src/kakarot/precompiles/test_precompiles.py

```diff
@@ -16,13 +16,58 @@ async def precompiles(starknet: Starknet):
 
 @pytest.mark.asyncio
 class TestPrecompiles:
-    class TestPrecompilesRun:
-        async def test_precompiles_should_throw_on_out_of_bounds(self, precompiles):
-            # we choose an out of range precompile address to get the NotImplementedPrecompile error, and test if it is including the address in the error
-            # note: in our implementation, `Precompiles.is_precompile` checks if an address is within a given range before dispatching, so usually an out of range address would never be passed to `Precompiles.run`
+    class TestRun:
+        async def test__run_should_return_a_stopped_execution_context(
+            self, precompiles
+        ):
+            # we choose datacopy precompile address
+            dataCopy_address = 0x4
+            await precompiles.test__run_should_return_a_stopped_execution_context(
+                address=dataCopy_address
+            ).call()
+
+    class TestIsPrecompile:
+        async def test__is_precompile_should_return_false_when_address_is_greater_than_last_precompile(
+            self, precompiles
+        ):
+            # we choose the address after the last precompile address
             last_precompile_address = 0x9
             address = last_precompile_address + 1
-            with kakarot_error("Kakarot: NotImplementedPrecompile " + str(address)):
+            await precompiles.test__is_precompile_should_return_false_when_address_is_greater_than_last_precompile(
+                address=address
+            ).call()
+
+        async def test__is_precompile_should_return_true_when_address_is_lower_than_last_precompile(
+            self, precompiles
+        ):
+            # we choose the address of an implemented precompile
+            address = 0x6
+            await precompiles.test__is_precompile_should_return_true_when_address_is_lower_than_last_precompile(
+                address=address
+            ).call()
+
+    class TestNotImplementedPrecompile:
+        async def test__not_implemented_precompile_should_raise_with_detailed_error_message(
+            self, precompiles
+        ):
+            # we choose an address of a non implemented precompile and we check that the Non Implemented Precompile error msg appear.
+            not_impl_precompile_address = 0x8
+            with kakarot_error(
+                "Kakarot: NotImplementedPrecompile " + str(not_impl_precompile_address)
+            ):
                 await precompiles.test__precompiles_should_throw_on_out_of_bounds(
-                    address=address
+                    address=not_impl_precompile_address
+                ).call()
+
+    class TestExecPrecompile:
+        async def test__exec_precompiles_should_throw_on_out_of_bounds(
```

Comment:
this is copy/pasted from the previous test. For this one just use `address = 0x9 + 1` and 

```
with kakarot_error():
    await ...
```

without testing the error message (there is no particular error, just should throw most often unknown memory location but can be something else)